src/main/java/com/faforever/client/teammatchmaking/PartyMemberItemController.java

```diff
@@ -137,6 +139,23 @@ private void setPlayerProperties() {
     });
   }
 
+  @VisibleForTesting
+  protected void setLeagueInfo() {
+    leaderboardService.getHighestLeagueEntryForPlayer(player).thenAccept(leagueEntry -> JavaFxUtil.runLater(() -> {
+      if (leagueEntry.isEmpty() || leagueEntry.get().getSubdivision() == null) {
+        leagueLabel.setText(i18n.get("teammatchmaking.inPlacement").toUpperCase());
+        leagueImageView.setVisible(false);
+      } else {
+        SubdivisionBean subdivision = leagueEntry.get().getSubdivision();
+        leagueLabel.setText(i18n.get("leaderboard.divisionName",
+            i18n.getOrDefault(subdivision.getDivision().getNameKey(), subdivision.getDivisionI18nKey()),
+            subdivision.getNameKey()).toUpperCase());
+        leagueImageView.setImage(leaderboardService.loadDivisionImage(subdivision.getMediumImageUrl()));
+        leagueImageView.setVisible(true);
+      }
```

Comment:
This is still loading the image inside of a JavaFxutil.runLater

Reply:
the other usages of `loadAndCacheImage` that I checked don't bother with loading the image outside of the application thread, so I thought the call to the service might switch the thread back.
So I load the image into a local variable and then only set that in the application thread?

Reply:
hmmm yeah I just realized that we force the loadAndCacheImage to load in the background for fetching fresh images but not for loading from cache. If you add the true flag to load from background when loading from cache as well then this should actually be fine